common/src/main/java/com/microsoft/identity/common/internal/broker/BoundServiceFuture.java

```diff
@@ -23,18 +23,24 @@
 
 package com.microsoft.identity.common.internal.broker;
 
-import com.microsoft.identity.client.IMicrosoftAuthService;
+import android.os.IInterface;
+
+import androidx.annotation.NonNull;
 
 import java.util.concurrent.CountDownLatch;
 import java.util.concurrent.ExecutionException;
 import java.util.concurrent.Future;
 import java.util.concurrent.TimeUnit;
 import java.util.concurrent.TimeoutException;
 
-public class MicrosoftAuthServiceFuture implements Future<IMicrosoftAuthService> {
+/**
+ * A Future object for bound service.
+ * Will block until a bound service connection is established.
+ */
+public class BoundServiceFuture implements Future<IInterface> {
```

Comment:
Can we just use a SettableFuture here?

Reply:
by importing com.google.guava:guava:28.1-android, yes.

Reply:
(although I don't think it's worth it for one-time use - since this will also inflate the customer's apk size).

Reply:
I was more thinking androidx.

Reply:
replaced BoundServiceFuture with common's ResultFuture<T>